<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ConfigWidget</class>
 <widget class="QWidget" name="ConfigWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>500</width>
    <height>500</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>WarpKate Configuration</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="generalTab">
      <attribute name="title">
       <string>General</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QGroupBox" name="behaviorGroupBox">
         <property name="title">
          <string>Behavior</string>
         </property>
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="shellLabel">
            <property name="text">
             <string>Shell Command:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="shellEdit">
            <property name="placeholderText">
             <string>/bin/bash</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="startupCmdLabel">
            <property name="text">
             <string>Startup Command:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="startupCmdEdit"/>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="QCheckBox" name="autoshowCheck">
            <property name="text">
             <string>Show terminal on startup</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0" colspan="2">
           <widget class="QCheckBox" name="saveHistoryCheck">
            <property name="text">
             <string>Save command history between sessions</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="historySizeLabel">
            <property name="text">
             <string>Command History Size:</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QSpinBox" name="historySizeSpinBox">
            <property name="minimum">
             <number>50</number>
            </property>
            <property name="maximum">
             <number>10000</number>
            </property>
            <property name="value">
             <number>1000</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="layoutGroupBox">
         <property name="title">
          <string>Layout</string>
         </property>
         <layout class="QFormLayout" name="formLayout_3">
          <item row="0" column="0">
           <widget class="QLabel" name="positionLabel">
            <property name="text">
             <string>Terminal Position:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="positionCombo">
            <item>
             <property name="text">
              <string>Bottom</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Left</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Right</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="heightLabel">
            <property name="text">
             <string>Default Height:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="heightSpinBox">
            <property name="suffix">
             <string>px</string>
            </property>
            <property name="minimum">
             <number>100</number>
            </property>
            <property name="maximum">
             <number>2000</number>
            </property>
            <property name="value">
             <number>300</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="appearanceTab">
      <attribute name="title">
       <string>Appearance</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QGroupBox" name="colorsGroupBox">
         <property name="title">
          <string>Colors</string>
         </property>
         <layout class="QFormLayout" name="formLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="colorSchemeLabel">
            <property name="text">
             <string>Color Scheme:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="colorSchemeCombo">
            <item>
             <property name="text">
              <string>System Default</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Dark</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Light</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Solarized Dark</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Solarized Light</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Custom</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="fontLabel">
            <property name="text">
             <string>Font:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="fontSelectButton">
            <property name="text">
             <string>Select Font...</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="QCheckBox" name="transparencyCheck">
            <property name="text">
             <string>Enable transparency</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="transparencyLabel">
            <property name="text">
             <string>Transparency Level:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QSlider" name="transparencySlider">
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="value">
             <number>20</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="blocksGroupBox">
         <property name="title">
          <string>Command Blocks</string>
         </property>
         <layout class="QFormLayout" name="formLayout_4">
          <item row="0" column="0">
           <widget class="QLabel" name="blockStyleLabel">
            <property name="text">
             <string>Block Style:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="blockStyleCombo">
            <item>
             <property name="text">
              <string>Modern</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Classic</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Compact</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QCheckBox" name="showTimestampsCheck">
            <property name="text">
             <string>Show command timestamps</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="QCheckBox" name="syntaxHighlightCheck">
            <property name="text">
             <string>Enable syntax highlighting for command output</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="aiTab">
      <attribute name="title">
       <string>AI Integration</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QGroupBox" name="aiGroupBox">
         <property name="title">
          <string>AI Settings</string>
         </property>
         <layout class="QFormLayout" name="formLayout_5">
          <item row="0" column="0" colspan="2">
           <widget class="QCheckBox" name="enableAICheck">
            <property name="text">
             <string>Enable AI-powered suggestions</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="aiModelLabel">
            <property name="text">
             <string>AI Model:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="aiModelCombo">
            <item>
             <property name="text">
              <string>Local (Fast)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Remote (Advanced)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Custom API</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="apiKeyLabel">
            <property name="text">
             <string>API Key:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="apiKeyEdit">
            <property name="echoMode">
             <enum>QLineEdit::Password</enum>
            </property>
            <property name="placeholderText">
             <string>Enter your API key here</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0" colspan="2">
           <widget class="QCheckBox" name="contextAwarenessCheck">
            <property name="text">
             <string>Enable context awareness (use current document for context)</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="4" column="0" colspan="2">
           <widget class="QCheckBox" name="privacyModeCheck">
            <property name="text">
             <string>Privacy mode (limit information sent to AI service)</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="suggestionGroupBox">
         <property name="title">
          <string>Suggestion Behavior</string>
         </property>
         <layout class="QFormLayout" name="formLayout_6">
          <item row="0" column="0" colspan="2">
           <widget class="QCheckBox" name="autoSuggestCheck">
            <property name="text">
             <string>Automatically show suggestions while typing</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="maxSuggestionsLabel">
            <property name="text">
             <string>Maximum Suggestions:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="maxSuggestionsSpinBox">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>10</number>
            </property>
            <property name="value">
             <number>3</number>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="suggestionDelayLabel">
            <property name="text">
             <string>Suggestion Delay:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="suggestionDelaySpinBox">
            <property name="suffix">
             <string>ms</string>
            </property>
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>2000</number>
            </property>
            <property name="singleStep">
             <number>100</number>
            </property>
            <property name="value">
             <number>500</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Apply|QDialogButtonBox::Cancel|QDialogButtonBox::Ok|QDialogButtonBox::Reset</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
